driver_test.go

```diff
@@ -338,3 +309,301 @@ func TestIsDdl(t *testing.T) {
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ), (2, "two", CAST("two" as bytes), 42, 42, true )`,
+			givenInput:  `DELETE FROM TestExecContextDml WHERE key = 1`,
+			whenQueried: `SELECT * FROM TestExecContextDml`,
+			wantRows: []testExecContextDmlRow{
+				{key: 2, testString: "two", testBytes: []byte("two"), testInt: 42, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name: "delete multiple tuples with subwhenQueried",
+			setUp: `
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ), (2, "two", CAST("two" as bytes), 42, 42, true ),
+				(3, "three", CAST("three" as bytes), 42, 42, true ), (4, "four", CAST("four" as bytes), 42, 42, true )`,
+			givenInput: `
+				DELETE FROM TestExecContextDml WHERE key
+				IN (SELECT key FROM TestExecContextDml WHERE key != 4)`,
+			whenQueried: `SELECT * FROM TestExecContextDml ORDER BY key`,
+			wantRows: []testExecContextDmlRow{
+				{key: 4, testString: "four", testBytes: []byte("four"), testInt: 42, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name: "delete all tuples",
+			setUp: `
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ), (2, "two", CAST("two" as bytes), 42, 42, true ),
+				(3, "three", CAST("three" as bytes), 42, 42, true ), (4, "four", CAST("four" as bytes), 42, 42, true )`,
+			givenInput:  `DELETE FROM TestExecContextDml WHERE true`,
+			whenQueried: `SELECT * FROM TestExecContextDml`,
+			wantRows:    []testExecContextDmlRow{},
+		},
+		{
+			name: "update one tuple",
+			setUp: `
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ), (2, "two", CAST("two" as bytes), 42, 42, true ),
+				(3, "three", CAST("three" as bytes), 42, 42, true )`,
+			givenInput:  `UPDATE TestExecContextDml SET testSTring = "updated" WHERE key = 1`,
+			whenQueried: `SELECT * FROM TestExecContextDml ORDER BY key`,
+			wantRows: []testExecContextDmlRow{
+				{key: 1, testString: "updated", testBytes: []byte("one"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 2, testString: "two", testBytes: []byte("two"), testInt: 42, testFloat: 42, testBool: true},
+				{key: 3, testString: "three", testBytes: []byte("three"), testInt: 42, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name: "update multiple tuples",
+			setUp: `
+				INSERT INTO TestExecContextDml (key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ), (2, "two", CAST("two" as bytes), 42, 42, true ),
+				(3, "three", CAST("three" as bytes), 42, 42, true )`,
+			givenInput:  `UPDATE TestExecContextDml SET testSTring = "updated" WHERE key = 2 OR key = 3`,
```

Comment:
I think we might be confusing the terms here:

- `given`: should be used to setup the test (adjust the expected state of the test). In this case it would be the `INSERT ...` statement.
- `when`: should be the command that we would like to test. In this case it would be the `UPDATE ...` statement.
- `then`: should be assertion we would like to make. In this case it would be `SELECT ...` statement and the wanted rows.

IMHO this would ready better like so:

```
{
  testCase: "update multiple tuples",
  given: "INSERT INTO ..."
  when: "UPDATE ..."
  then: {
    query: "SELECT ..."
    expectedRows: []testExecContextDmlRow{}
  }
}
```

Reply:
Changed, must have been confused because most of the tests don't use setUp/given but it still seems to make sense with without it